### PR TITLE
[ISSUE-77] Add token module to dependencies.

### LINKED ISSUE #77: Quant Search fails silently if tokens are used that aren't available
See details in drupal.org issue:

https://www.drupal.org/project/quantcdn/issues/3323931

### PR DESCRIPTION
**Pull request by kepol**

### CODE REVIEW

**Comment by kepol** [comment]
I think I have finally tweaked the logic to work for all scenarios including empty tokens and things like:

```
[
]
[]
[:
:]
[:]
[blah
[blah:
[blah:]
[blah:]more
[blah:foo
[blah:foo]
[blah:foo]more
foo]
:foo]
[:foo]
[:foo]more
ugh
ugh[
ugh[blah
ugh[blah:
ugh[blah:foo
ugh[blah:foo]
ugh[blah:foo]more
]more
foo]more
:foo]more
blah:foo]more
[blah:foo]more
ugh[blah:foo]more
ugh[blah:foo]more[blah:foo]more
```

**Comment by kepol** [comment]
Example errors:

Screen Shot 2022-11-29 at 9 19 32 PM
Screen Shot 2022-11-29 at 9 19 42 PM
Screen Shot 2022-11-29 at 9 19 47 PM
Screen Shot 2022-11-29 at 9 19 54 PM

**Comment by kepol** [comment]
This needs review... the relevant code is in `validateForm` and the `quant.info.yml`. The rest of the code changes are just rejiggering the order/labels of the form fields so it's less confusing.

File: quant.info.yml [hunk 1]
```diff
@@ -7,4 +7,5 @@ core_version_requirement: ^8 || ^9
 core: 8.x
 
 dependencies:
+  - drupal:token
```
Comment (stooit) [inline quant.info.yml]: I think technically this dependency should go into `quant_search` -- the main quant module works fine without token

File: modules/quant_search/src/Form/SearchEntitiesForm.php [hunk 2]
```diff
@@ -242,9 +244,100 @@ public function buildForm(array $form, FormStateInterface $form_state) {
...
+      // Only check tokens if default or node type is overriding the defaults.
+      if ($nodeTypeId === 'default' || $typeTokens['enabled']) {
+        $tokens[] = trim($typeTokens['quant_search_title_token']);
+        $tokens[] = trim($typeTokens['quant_search_summary_token']);
+        $tokens[] = trim($typeTokens['quant_search_image_token']);
+      }
+    }
+
+    // Token module will find some of the invalid tokens.
+    $invalidTokens = \Drupal::token()->getInvalidTokensByContext(implode(' ', $tokens), ['all']);
```
Comment (stooit) [inline modules/quant_search/src/Form/SearchEntitiesForm.php:293]: We may need a `quant_search` update hook to enable the token module for a clean upgrade pathway.

If someone were to update an existing install without token installed they would get `Call to undefined method Drupal\Core\Utility\Token::getInvalidTokensByContext()` errors

File: quant.info.yml (hunk 1 above)
Comment (kepol) [reply quant.info.yml]: Good point, I'll move.

File: modules/quant_search/src/Form/SearchEntitiesForm.php (hunk 2 above)
Comment (kepol) [reply modules/quant_search/src/Form/SearchEntitiesForm.php:293]: Good catch, I'll add.

**Comment by kepol** [comment]
I made the requested changes and tested as follows and the update hook worked as expected.

1. no token module in code base

```
Kristens-MacBook-Pro:quantcdn kristenpol$ ddev drush updb
 -------------- ----------- --------------- ----------------------------- 
  Module         Update ID   Type            Description                  
 -------------- ----------- --------------- ----------------------------- 
  quant_search   8102        hook_update_n   8102 - Enable token module.  
 -------------- ----------- --------------- ----------------------------- 

 Do you wish to run the specified pending updates? (yes/no) [yes]:> 

>  [notice] Update started: quant_search_update_8102
>  [warning] Undefined array key "token" ModuleInstaller.php:125
>  [warning] Attempt to read property "info" on null ModuleInstaller.php:125
>  [warning] Trying to access array offset on value of type null ModuleInstaller.php:125
>  [warning] Undefined array key "token" ModuleInstaller.php:128
>  [warning] Attempt to read property "info" on null ModuleInstaller.php:128
>  [warning] Trying to access array offset on value of type null ModuleInstaller.php:128
>  [error]  Unable to install modules token due to missing modules token. 
>  [error]  Update failed: quant_search_update_8102 
 [error]  Update aborted by: quant_search_update_8102 
 [error]  Finished performing updates. 
Failed to run drush updb: exit status 1

```

2. token in code base but not installed 

```
Kristens-MacBook-Pro:quantcdn kristenpol$ ddev drush updb
 -------------- ----------- --------------- ----------------------------- 
  Module         Update ID   Type            Description                  
 -------------- ----------- --------------- ----------------------------- 
  quant_search   8102        hook_update_n   8102 - Enable token module.  
 -------------- ----------- --------------- ----------------------------- 

 Do you wish to run the specified pending updates? (yes/no) [yes]:> 

>  [notice] Update started: quant_search_update_8102
>  [notice] The token module has been enabled.
>  [notice] Update completed: quant_search_update_8102
>  [notice] Checked es translation for token.
>  [notice] Downloaded es translation for token.
>  [notice] Imported es translation for token.
>  [notice] Translations imported: 0 added, 182 updated, 0 removed.
>  [warning] No configuration objects have been updated.
```

3. token already installed

```
Kristens-MacBook-Pro:quantcdn kristenpol$ ddev drush updb
 -------------- ----------- --------------- ----------------------------- 
  Module         Update ID   Type            Description                  
 -------------- ----------- --------------- ----------------------------- 
  quant_search   8102        hook_update_n   8102 - Enable token module.  
 -------------- ----------- --------------- ----------------------------- 

 Do you wish to run the specified pending updates? (yes/no) [yes]: > 

>  [notice] Update started: quant_search_update_8102
>  [notice] The token module is already enabled.
>  [notice] Update completed: quant_search_update_8102
```

**Review by stooit — APPROVED** [review]

**Pull request merged**